test/functional-tests/index.js

```diff
@@ -383,4 +391,118 @@ describe('tests', () => {
         }));
     });
   });
+
+  describe('#Postcode-Lookup', () => {
+    describe('default postcode lookup behaviour', () => {
+      before(() => {
+        app = App(require('./apps/postcode-lookup-default')({ port })).listen(port);
+        port = app.address().port;
+      });
+
+      after(() => {
+        app.close();
+      });
+
+      it('stays on the postcode entry page if the postcode input is left as blank', () => browser.url('/postcode-default-one')
+        .$('input')
+        .setValue('')
+        .submitForm('form')
+        .getUrl()
+        .then(url => {
+          assert.ok(url.includes('postcode-default-one'));
+        }));
+
+      it('stays on the postcode entry page if the entered postcode is of an invalid format', () => browser.url('/postcode-default-one')
```

Comment:
Would also check if validation error message is shown

Reply:
Included this scenario in the unit tests.